### DISCUSSION TITLE
unneccessary openning of valves after fast and high temperature rise due to window/door state change

### DISCUSSION THREAD

**Discussion by harvey637** [Lenti84/VdMot_Controller#89] [Ideas]
Hi @Lenti84 and @Gargano
As observed daily here some more info on this topic:

![grafik](https://user-images.githubusercontent.com/10156426/236665445-9fd486dd-c8c2-4094-ab64-868a62fd52e7.png)

- special code for "unnatural" rise in temperature? (fast fall in temperature is handled ok, as I want NO heating immediately at that time).
- no need for extra input of sensors, as the rise is sognificant and can be detected (easily)?
- with unnatural I mean, that such a rise in that short time can not be change in weather or heating demand with floor heating

When I speak of heat demand I can imaging calculation of the demand in vdmot_controller and giving start/stop signal to heatpump. Now the heatpump is working completly independant of the vdmot. Just keep in mind that there is also cooling demand :-) see you in summer!

thank you
and ciao
Harvey

**Comment by SurfGargano** [comment]
Can you pls show me your control setup

**Reply by harvey637** [reply]
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/22aba86e-6780-4b19-a39e-5aae4d8affbb)

All the control comes from temperature sensors, where a change of the measured temperature OR the set temperature triggers a homematicscript like this:
```
room_temp = (dom.GetObject(ID_CHANNELS)).Get(thermostat#":1").DPByHssDP("ACTUAL_TEMPERATURE").Value();
set_temp  = (dom.GetObject(ID_CHANNELS)).Get(thermostat#":1").DPByHssDP("SET_POINT_TEMPERATURE").Value();

if (debug >= 1) {
  WriteLine("Raumtemperatur "#thermostat#" (Ventil: "#valve_nr#"/"#valve#") ist "#room_temp.ToString(1)#"°C, Solltemperatur: "#set_temp.ToString(1)#"°C.");
  ! dom.GetObject("CUxD.CUX2801001:1.SYSLOG").State("Raumtemperatur "#thermostat#" (Ventil: "#valve_nr#"/"#valve#") ist "#room_temp.ToString(1)#"°C, Solltemperatur: "#set_temp.ToString(1)#"°C.");
}

if (debug < 2) {

  system.Exec ("/usr/local/addons/solaredge/vdmot-ctrlValve "#valve_nr#" "#room_temp#" "#set_temp#" ", &stdout, &stderr);
  if (debug == 1) {
    WriteLine("std: "#stdout);
    WriteLine("err: "#stderr);
  }


```

The call Exec is simply a call to the vdmot-api pushing the values to vdmot-controller ....
So the vdmot controller is called with the change of the set temperature caused by the open window or door. In the chart you see a small ofsset of the real opening/closing to the set_temperature change in the thermostat. This is to prevent from triggering with very short window changes, as to clsoe between half closed/tilted into full open.
ciyao
Harvey

**Reply by harvey637** [reply]
Ups, what I never saw before: the valve nr 9 is "lost" in status ?!?!? It is still configured and attached, I did not change anything with the valves (but the repair...)

**Reply by harvey637** [reply]
after "SCAN" and some time they are all back again!
AFTER your holiday I can show you more screenshots, if you want more ...
ciao
Harvey

**Reply by SurfGargano** [reply]
But you disconnect the valve to repair , right? So it is correct, that the valve disappeared. You should get a notification via pushover or email if activated.

**Reply by harvey637** [reply]
Hi,
sadly no :-( I repaired valve 8, and after repair I did several calibrations to get the pushing part into correct position. Also I moved the repaired valve from "Bad" (where it is moved quite often) to a seldom used position (KleinesBad). So valves were exchanged too, and calibrated after that. "Küche=valve9" was connected all the time (but updates and reboots too happened).
So valve9 vanished "out of the blue".

Funnily I cannot find the time when it was lost, as it appears that it was used all the time, or it was immedeately before I took the screenshot. At least the api-calls look like the valve was reported,but I dont do extensive error checking there. As it is getting (a little bit) warmer valves are long time in position 0, so this is also the default return from my api call, even if the regex does not find the searched string in the api return.
I found that only "active" valves are part of the api return json, so there is no "unconnected" state for an unconnected (and/or unconfigured) valve. Maybe a state of "unconfigured" or "stale" with a position "0" for ALL 12 valves ??? or position "-1"??? Just thinking... Might also be a variant of the status windows (url.....?show_all) with my wish for a seperate status page without user/password :-)

```
   set url "http://vdmot/valves"
   exec /usr/bin/wget --no-check-certificate -q -O /usr/local/addons/solaredge/vdmot_valves.json $url
...
   set kleinesbad 0 ;
   set az_hawe 0 ;
   set bad 0 ;
   set kueche 0 ;
...
# --- Küche
    regexp "\"K..che\",\"state\":(.*?),\"pos\":(.*?)," $input dummy state kueche ;
#    puts "Küche: $kueche"

# --- Bad
    regexp "\"Bad\",\"state\":(.*?),\"pos\":(.*?)," $input dummy state bad ;
#    puts "Bad:  $bad"
...
```
(this was part of my old question to fetch a single valve state/position by api-call, as this very small script does no do full json parsing, but I can live with it perfectly now)

No, as of now I dont do emails from vdmot-controller, as I have no idea how to put emails INTO my smarthome, so I have to live with (possibly better) error checking inside homematic scripts. Maybe I should check "status" also to get ... "idle/connected/opens/closes/????"
What is the state of an unconnected (or lost) valve? Will test sometime :-)
ciao
Harvey

**Reply by SurfGargano** [reply]
The state of unconnect valve is open circuit. The notification is also made via mqtt. I use the pushover function. It costs about 5 euros one time. I use it not only for Vdmot, but also for my worx landroid and other things that can hapoen in my home.

**Comment by SurfGargano** [comment]
For setting all valves control at a time pls see https://github.com/Lenti84/VdMot_Controller/wiki/JSON-Interface
This is implemented in V1.2.1 on my github

**Comment by SurfGargano** [comment]
Hi harvey637,
coming back to the problem of unnatural rising of target temparature. I think, that could be done only by a special code, apart of the regular code. If the target temperature is dropped down dramatically (in case of open a window or door) and coming back with dramatically rise , then the valve value remain at the current value and start from this point to control the valve without new calculating with the proportional part.

I have put hose clamps (32-50mm) on my homematic motors to prevent jumping off.

btw . On my git is the last V1.2.2 with some new features, bugfixes and a modified (just modern) web appearance.

**Comment by harvey637** [comment]
this is some of my valves:
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/fc04ef25-5bb8-4bd6-9dca-e10f66ffd464)

even better here:
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/febfde05-6623-4fd6-ae39-f9353d7e1f25)
you can see the red line (room temp) ALLWAYS minimum around 1°C higher than requested temperature, no need to open valve.

Opening is about 15% and goes back to 0% in about 30 minutes.
Grid is x=1 day, y=5°C

**Comment by harvey637** [comment]
My guess is, that TARGET temperature is allways "jumping" as going from night_temperature_down to day_temperature_up. In my case (second graph): night and working_hours: 17°C, morning and evening: 20,5°C. 
As the room temperature is sufficient above the requested level at any time there should be no need to open the valve.
This is true more or less the whole summer, as there is no heating with regular summer temperatures in the house.

The only movement of valves should be the weekly calibration (where I saw that you are modifying code around this).

no 1.2.2 release yet found :-(

**Reply by SurfGargano** [reply]
Rel 1.2.2 is on my git https://github.com/SurfGargano/VdMot_Controller/tree/developer/releases/1.2.2_Dev

![image](https://github.com/Lenti84/VdMot_Controller/assets/70272783/0759fd4d-3832-454b-87c0-bc32b00af30f)

**Comment by SurfGargano** [comment]
For this reason the park position is useful :
![image](https://github.com/Lenti84/VdMot_Controller/assets/70272783/d07dde1b-aa52-4f07-a21a-0483eafdb12d)

**Comment by harvey637** [comment]
just my point of view:
- the heatpump is switched on all around the year, as there might be "very" cold nights in summer (heating some hours in the morning)  and some very hot days with automatically cooling (not this year yet).
So I do not handle the valves from outside with extra code (putting them in to parking mode and OUT of parking mode), as the target temperature allways signals that no valve opening is needed (on normal summer days, we had a few (2~3) very cold nights with heating and wanted valve openings).

So my point of view is: 365 days position the valves according to the diffence of room_temp to  target_temperature: if room_temp equal or below target_temp open valve in heating_mode open the valve.
just my idea :-)
I thought of parking position to detache the motor from the valve at specific position ...

**Reply by SurfGargano** [reply]
_So my point of view is: 365 days position the valves according to the diffence of room_temp to target_temperature: if room_temp equal or below target_temp open valve in heating_mode open the valve._

That is what the VDMot is doing, except your problem with the extrem jump. But you can play with the parameter xp to reduce the jump.
xp is the parameter for the proportional part.

**Comment by harvey637** [comment]
ok, updated: VERY short first feedback:
- nice on/off slider (but color is not harmonic to other used colors, my personal taste :-)
- details on status are infomative
- IMHO status should be "view_only", the manual movement of valves does not belong there (as beeing view_only no user/pwd is needed than).
- I think that manual control of valves should belong to "control" page ...

thank you
Harvey

**Reply by SurfGargano** [reply]
a new version of 1.2.2 is on my git : colors of the sliders are now the same as the buttons. If a login-user exists the status page is also shown. The activate-set slider is disabled in this case.

**Comment by harvey637** [comment]
more on the topic of "extreme" jump:
if you check the first of my last screenshots you can finnd very small changes in target temperature from 20.0°C to 20,5°C, with room temperature around 21°C ..... this is no extreme jump :-)
for more information see next screenshot:
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/ec8bccd8-a0e1-4354-9797-5a799fb9897c)

during the whole time heatpump was OFF, room_temperature ALLWAYS above target_temperature .... and so many small and short valve moves :-)
I am not complaining, just offering information and time for a discussion! thank you
Harvey

**Reply by SurfGargano** [reply]
Can you pls show the controlpage

**Reply by harvey637** [reply]
yes, Sir!
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/425c59f7-dce7-42fc-9c19-c5f62d627c30)

**Comment by harvey637** [comment]
Hi @ Surfgargano,
thank you for the upupdate 1.2.2_dev.
- colors look nice
- status page without login is great .... what about removing the activate-set slider  and the up/down completely? (minor-minor-minor:-)
- some overlapping in config page
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/65958da0-93cb-43cd-b595-4ce690d99e6e)

- DNS is configured but not shown
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/96f50c9e-9f85-4128-84d3-b360eab5aae8)

Status page:
- moves are incredibly high, possible int32 -1  ? => set to 0 on first use :-)
- what is dead count?
- possibly show last direction (- | open | close ) (feature request with no! prio)

- all dates shown look a bit strange (hmmmmmm, they are ok, but ....)
- the wording of the update/home page cold be the same as the filenames (e.g. ESP32 update), slightly confusing
- columns in control page too small for some text (heat/cool, dynamic ki)
cu
Harvey

**Reply by SurfGargano** [reply]
DNS like the other info values in the network page is not the configured but the real existing values. So in your case dns is not existent.
Hiding the activate-set slider -> comes with next version.
Overlap comes with firefox, with chrom it is ok. I have to check.

**Reply by harvey637** [reply]
thank you, perfect, dont hurry, keep your time!
DNS: I have DNS configured and my router (fritz box) is running as internal dns-proxy. So it should work?!?
Other systems as raspberrymatic or iobroker show it.

![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/0f815210-7c77-4d26-8959-3b02a1c7ddfa)


(sorry for updating my answer)
cu
Harvey

**Reply by SurfGargano** [reply]
Move looks strange seems to be -1, You have update stm with the latest one ?

**Reply by harvey637** [reply]
yesterday wt32 + stm32, today only wt32 (esp32) as date of stm is unchanged ....

**Reply by SurfGargano** [reply]
Okay, I see a problem with stm, could you change one valve to another position. Then the move should show the value 1

What is the movements in the config / motor ?

**Reply by harvey637** [reply]
![grafik](https://github.com/Lenti84/VdMot_Controller/assets/10156426/ddbc6e19-a502-4c4a-8c30-9b8071a7d064)

**Reply by harvey637** [reply]
even funny, after rebooting the counts go lower :-) e.g. 
last line 8493 now 8486,

**Reply by SurfGargano** [reply]
ok, next try. I have fixed (hopefully) the bug with the moves. In my git are the files.

**Reply by harvey637** [reply]
yes, all moves start with 0 - great!

by the way, I bookmarked the "new" vdmot/#status page in my browser, but is stiil goes to /#home instead of 7#status .... still checking ....

**Comment by SurfGargano** [comment]
Open Count , Close Count are the counts detected of the motor, Dead Count is Open Count - Close Count.

**Comment by SurfGargano** [comment]
Dear Harvey637,
regarding your problem with opening valves : 
on my git there exists a new version : 
https://github.com/SurfGargano/VdMot_Controller/tree/developer/releases/1.2.5_Dev
Latest Version : 
WTH32 1.2.5
STM 1.2.2
It includes a window handling. Pls see at 
https://github.com/Lenti84/VdMot_Controller/wiki/Web-UI#control
for description.

Also it exists a new telegram chat : 
https://t.me/+P0_VMEDQ8NMzZTUy

**Comment by harvey637** [comment]
Hi surfGargano,
happy to see you working on this still ..... i will install this version and give feedback.

The window feature is requested, ok, but uninteresting to me. All my smart stuff like changing temperature with open doors/windows is handled in my "main" system homematic+iobroker.

Also  - just a reminding question - have you considered, that in HEATING an open windows LOWERS temperature (what will be the temperature drop?) to prevent from unneccessary heating the outside environment. In COOLING an open window should RAISE the temperature, as warm air will get in from outside, thus it makes no sense lowering the temperature, I even raise the temperature.
Example: Cooling, outside temp 37°C, inside cooling to 32°C. Open the door raises to plus_4°C (=36°C), so no more cooling for this room, valve gets closed. 
cu
Harvey

**Reply by SurfGargano** [reply]
With the new window function you can define the valve setting when the window is open. So for example  set the target in window to 20 (%) and the set the state to 1. The valve goes to 20 % and don't regulate while the window is open. Pls note that the target must be set prior setting the state.  So it is up to you what the target is set while window is open.

**Reply by harvey637** [reply]
ok, understood and saw it also in your documentation.

updated vdmot, now waiting for some valve movement and will feedback the next days ...
thank you and cu
Harvey
